L1Trigger/L1TTrackMatch/plugins/L1TkGlbMuonProducer.cc

```diff
@@ -0,0 +1,316 @@
+      // a pT dependent dR window
+      if (dr2prop < drmin) {
+        drmin = dr2prop;
+        match_idx = il1tk;
+        matchProp = pstate;
+      }
+    }  // over l1tks
+
+    LogDebug("L1TkGlbMuonProducer") << "matching index is " << match_idx;
+    if (match_idx >= 0) {
+      const L1TTTrackType& matchTk = l1tks[match_idx];
+
+      float sigmaEta = sigmaEtaTP(*l1mu);
+      float sigmaPhi = sigmaPhiTP(*l1mu);
+
+      float etaCut = matching_factor_eta * sqrt(sigmaEta * sigmaEta + matchProp.sigmaEta * matchProp.sigmaEta);
+      float phiCut = matching_factor_phi * sqrt(sigmaPhi * sigmaPhi + matchProp.sigmaPhi * matchProp.sigmaPhi);
+
+      float dEta = std::abs(matchProp.eta - l1mu_eta);
+      float dPhi = std::abs(deltaPhi(matchProp.phi, l1mu_phi));
+
+      bool matchCondition = useTPMatchWindows_ ? dEta < etaCut && dPhi < phiCut : drmin < dRMax_;
+
+      if (matchCondition) {
+        edm::Ptr<L1TTTrackType> l1tkPtr(l1tksH, match_idx);
+
+        const auto& p3 = matchTk.momentum();
+        float p4e = sqrt(mu_mass * mu_mass + p3.mag2());
+
+        math::XYZTLorentzVector l1tkp4(p3.x(), p3.y(), p3.z(), p4e);
+
+        const auto& tkv3 = matchTk.POCA();
+        math::XYZPoint v3(tkv3.x(), tkv3.y(), tkv3.z());  // why is this defined?
+
+        float trkisol = -999;
+
+        TkGlbMuon l1tkmu(l1tkp4, l1muRef, l1tkPtr, trkisol);
+
+        l1tkmu.setTrkzVtx((float)tkv3.z());
+        l1tkmu.setdR(drmin);
+        l1tkmu.setNTracksMatched(nTracksMatch);
+        tkMuons.push_back(l1tkmu);
+      }
+    }
+  }  //over l1mus
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void L1TkGlbMuonProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+  desc.setUnknown();
+  descriptions.addDefault(desc);
+}
+
+L1TkGlbMuonProducer::PropState L1TkGlbMuonProducer::propagateToGMT(const L1TkGlbMuonProducer::L1TTTrackType& tk) const {
+  auto p3 = tk.momentum();
+  float tk_pt = p3.perp();
+  float tk_p = p3.mag();
+  float tk_eta = p3.eta();
+  float tk_aeta = std::abs(tk_eta);
+  float tk_phi = p3.phi();
+  float tk_q = tk.rInv() > 0 ? 1. : -1.;
+  float tk_z = tk.POCA().z();
+  if (!correctGMTPropForTkZ_)
+    tk_z = 0;
+
+  L1TkGlbMuonProducer::PropState dest;
+  if (tk_p < min_mu_propagator_p)
+    return dest;
+  if (tk_aeta < 1.1 && tk_pt < min_mu_propagator_barrel_pT)
+    return dest;
+  if (tk_aeta > max_mu_propagator_eta)
+    return dest;
+
+  //0th order:
+  dest.valid = true;
+
+  float dzCorrPhi = 1.;
+  float deta = 0;
+  float etaProp = tk_aeta;
+
+  if (tk_aeta < 1.1) {
+    etaProp = 1.1;
+    deta = tk_z / 550. / cosh(tk_aeta);
+  } else {
+    float delta = tk_z / 850.;  //roughly scales as distance to 2nd station
```

Comment:
The following lines of code seems a bit strange on the way of coding. Can it be simplify, less duplication of `if `and definition of `tk_z/850.`;